File: backend/__SLUG__/src/domains/testapi/controller.go
package testapi

import (
	"encoding/json"
	"net/http"
	"os"
)

// Controller handles test-only HTTP requests.
type Controller struct {
	service *Service
}

// NewController creates a new test API controller.
func NewController(service *Service) *Controller {
	return &Controller{service: service}
}

type deleteUsersRequest struct {
	IDs []string `json:"ids"`
}

type deleteUsersResponse struct {
	Deleted int `json:"deleted"`
}

// DeleteUsers deletes a batch of users from SuperTokens and the users table.
func (c *Controller) DeleteUsers(w http.ResponseWriter, r *http.Request) {
	apiKey := os.Getenv("TEST_API_KEY")
	if apiKey == "" || r.Header.Get("X-Test-Api-Key") != apiKey {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusForbidden)
		json.NewEncoder(w).Encode(map[string]string{"error": "forbidden"})
		return
	}

	var req deleteUsersRequest
	if err := json.NewDecoder(r.Body).Decode(&req); err != nil || len(req.IDs) == 0 {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusBadRequest)
		json.NewEncoder(w).Encode(map[string]string{"error": "invalid request body"})
		return
	}

	deleted, err := c.service.DeleteUsers(r.Context(), req.IDs)
	if err != nil {
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusInternalServerError)
		json.NewEncoder(w).Encode(map[string]string{"error": "internal server error"})
		return
	}

	w.Header().Set("Content-Type", "application/json")
	w.WriteHeader(http.StatusOK)
	json.NewEncoder(w).Encode(deleteUsersResponse{Deleted: deleted})
}

File: backend/__SLUG___node/src/domains/testapi/controller.ts
import type { Request, Response } from "express";
import type { Service } from "./service";

export class Controller {
  constructor(private readonly service: Service) {}

  /** Deletes a batch of users from SuperTokens and the users table. Gated by X-Test-Api-Key. */
  deleteUsers = async (req: Request, res: Response): Promise<void> => {
    const apiKey = process.env.TEST_API_KEY;
    if (!apiKey || req.header("X-Test-Api-Key") !== apiKey) {
      res.status(403).json({ error: "forbidden" });
      return;
    }

    const ids: unknown = req.body?.ids;
    if (!Array.isArray(ids) || ids.length === 0) {
      res.status(400).json({ error: "invalid request body" });
      return;
    }

    try {
      const deleted = await this.service.deleteUsers(ids as string[]);
      res.status(200).json({ deleted });
    } catch {
      res.status(500).json({ error: "internal server error" });
    }
  };
}
